commit 9d6b653c050973971b1ab8791d445b692871ab0c
Author: Jean-Christophe Dubois <jcd@tribudubois.net>
Date:   Sun Jul 1 00:10:50 2018 +0200

    thermal: i.MX: Allow thermal probe to fail gracefully in case of bad calibration.
    
    [ Upstream commit be926ceeb4efc3bf44cb9b56f5c71aac9b1f8bbe ]
    
    Without this fix, the thermal probe on i.MX6 might trigger a division
    by zero exception later in the probe if the calibration does fail.
    
    Note: This linux behavior (Division by zero in kernel) has been triggered
    on a Qemu i.MX6 emulation where parameters in nvmem were not set. With this
    fix the division by zero is not triggeed anymore as the thermal probe does
    fail early.
    
    Signed-off-by: Jean-Christophe Dubois <jcd@tribudubois.net>
    Reviewed-by: Fabio Estevam <fabio.estevam@nxp.com>
    Signed-off-by: Eduardo Valentin <edubezval@gmail.com>
    Signed-off-by: Sasha Levin <alexander.levin@microsoft.com>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

diff --git a/drivers/thermal/imx_thermal.c b/drivers/thermal/imx_thermal.c
index 334d98be03b9..b1f82d64253e 100644
--- a/drivers/thermal/imx_thermal.c
+++ b/drivers/thermal/imx_thermal.c
@@ -604,7 +604,10 @@ static int imx_init_from_nvmem_cells(struct platform_device *pdev)
 	ret = nvmem_cell_read_u32(&pdev->dev, "calib", &val);
 	if (ret)
 		return ret;
-	imx_init_calib(pdev, val);
+
+	ret = imx_init_calib(pdev, val);
+	if (ret)
+		return ret;
 
 	ret = nvmem_cell_read_u32(&pdev->dev, "temp_grade", &val);
 	if (ret)
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1134_9d6b653c050973971b1ab8791d445b692871ab0c.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1134_9d6b653c050973971b1ab8791d445b692871ab0c.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
